### ISSUE TITLE
Comment at replaceable type with constraining clause

### ISSUE THREAD

**Issue by abuntrock** [OpenModelica/OpenModelica#10030]
### Description

In the Modelica specification section 7.3.2.1 it is specified that the description of a replaceable declaration is recommended to be behind the constrainedby keyword. The omc and thus OMedit, however, seem to only support it, when the description is before the constrainedby keyword (which is identical according to the specification). In #9863 this was fixed for replaceable instances (second part of example in Specification section 7.3.2.1) but the problem still exist for replaceable types like `replaceable model` (first part of example in specification section 7.3.2.1).

### Steps to Reproduce

Consider the following test package below. The model _TestModel_ has one replaceable model declaration (CommentAfterConstraint) where the description is at the recommend position according to the specification and another replaceable model declaration (CommentBeforeConstraint) where the description is before the constrainedby keyword.

```modelica
package TestPackage
  model TestModel "TestModel comment"
  
  replaceable model CommentAfterConstraint = TestPackage.TestModel2 constrainedby TestPackage.TestModel2 "Replaceable model with comment after constraint"
  annotation (Dialog(group="Configuration parameters"));
  
  replaceable model CommentBeforeConstraint = TestPackage.TestModel2 "Replaceable model with comment before constraint" constrainedby    TestPackage.TestModel2 
  annotation (Dialog(group="Configuration parameters"));
  
  end TestModel;

  model TestModel2 "TestModel2 comment"
  end TestModel2;
end TestPackage;
```

### Expected Behavior

When calling `getClassComment(TestPackage.TestModel.CommentBeforeConstraint)` you get the following correct output: `"Replaceable model with comment before constraint"`.

But when calling `getClassComment(TestPackage.TestModel.CommentAfterConstraint)` the output is an empty string instead of the expected `"Replaceable model with comment after constraint"`. 

This also result in the fact that the comment of _CommentAfterConstraint_ is not shown in the paramter dialog of an instance of _TestModel_.

### Version and OS

 - OpenModelica Version: 1.21 dev
 - OS: Windows 10, 64 bit

Thanks for looking into it. 
Aaron Buntrock - Bosch rexroth

**Comment by adeas31** [comment]
@abuntrock the new instance api doesn't `getClassComment`. Do you also want that `getClassComment` is fixed as well?

**Comment by adeas31** [comment]
@perost I will take care of the OMEdit part while fixing #10140. Can you please fix `getClassComment`?

**Comment by abuntrock** [comment]
For us the priority is that it works correctly with the new API. However, we use `getClassComment` as part of our own CI testing. So if it's not too much to ask, we would like to have `getClassComment` working as well-

**Comment by abuntrock** [comment]
@adeas31 I just wanted to let you know that with todays nightly (v1.21.0-dev-253-g9bf056631b (64-bit)) in the new API both complete replaceable declaration are not shown in the parameter dialog anymore. Should I open a new ticket for this bigger new issue or is this just an intermediate state?

**Comment by adeas31** [comment]
There is no need for the new ticket. We know what the issue is and are working on it. #10140 is also the same issue.

**Comment by abuntrock** [comment]
That's right. I got confused between old and new API there. Sorry for that.
